2-Pricing/src/components/Modal.tsx

```diff
@@ -0,0 +1,58 @@
+import type { Cart } from "../App";
+import { Subscription } from "../constants";
+import { Backdrop } from "./Backdrop";
+
+interface ModalProps {
+  setIsModalVisible: (visible: boolean) => void;
+  subscriptionCandidate: Subscription;
+  inCart: Cart;
+  setInCart: (cart: Cart) => void;
+}
+
+export const Modal = ({
+  setIsModalVisible,
+  subscriptionCandidate,
+  inCart,
+  setInCart,
+}: ModalProps) => {
+  const addToCart = (subscription: Subscription) => {
+    const cartKey = subscription.name.toLowerCase() as keyof Cart;
+    const quantity = inCart[cartKey] ?? 0;
+    const updatedCart = { ...inCart };
+    updatedCart[cartKey] = quantity + 1;
+    setInCart(updatedCart);
```

Comment:
Grab `inCart` from the parameter of `setInCart`

Reply:
I would consider passing `addToCart` as a prop to `Modal`, and declaring this function in the parent component, instead of passing `inCart` and `setInCart`, since the modal can only add cart items.